app/src/main/assets/native/externalplayer.js

```diff
@@ -190,18 +190,20 @@ define(['appSettings', 'events', 'playbackManager'], function (appSettings, even
 
                     profile.SubtitleProfiles = [];
 
-                    var subtitleProfiles = ['ass', 'idx', 'pgs', 'pgssub', 'smi', 'srt', 'ssa', 'subrip'];
+                    var subtitleProfiles = ['ass', 'idx', 'smi', 'srt', 'ssa', 'subrip'];
 
-                    subtitleProfiles.forEach(function (format) {
+                    var embedSubtitleProfiles = ['pgs', 'pgssub'];
+
+                    subtitleProfiles.concat(embedSubtitleProfiles).forEach(function (format) {
                         profile.SubtitleProfiles.push({
                             Format: format,
                             Method: 'Embed'
                         });
                     });
 
-                    var externalSubtitleProfiles = ['srt', 'sub', 'subrip', 'vtt'];
+                    var externalSubtitleProfiles = ['sub', 'vtt'];
 
-                    externalSubtitleProfiles.forEach(function (format) {
+                    subtitleProfiles.concat(externalSubtitleProfiles).forEach(function (format) {
```

Comment:
You're now adding 2 profiles for all formats in `subtitleProfiles`: embed & external. I'm not sure if the server desires multiple methods for the same format.

Reply:
Yes, this is allowed, for example originally only `.ass` subtitles are allowed inside a `.mkv` file, if an external `.ass` file is added together with the `.mkv` file and is selected by the server/user, it forces transcoding, like in #192.


```
Movie.mkv
- ASS Embed Subtitle 1 <- SELECTED
- ASS Embed Subtitle 2
- ASS Embed Subtitle 3
Movie.en.default.ass
- ASS External Subtitle 4
```
Result: isEligibleForDirectStream: True

```
Movie.mkv
- ASS Embed Subtitle 1
- ASS Embed Subtitle 2
- ASS Embed Subtitle 3
Movie.en.default.ass
- ASS External Subtitle 4 <- SELECTED
```
Result: isEligibleForDirectStream: False
Not eligible for DirectStream due to unsupported subtitles

With this, both cases are allowed.

Reply:
Thanks for explaining!